_source/_posts/2022-01-19-react-storybook.md

```diff
@@ -0,0 +1,688 @@
+
+Now Storybook provides a framework that lets you test UI components in isolation from the rest of the website. Storybook presents you with a browser of all the components in your web application. You can test each component independently and in different configurations. The tool all runs as a separate application outside your main application, which means that you can test your components without worrying about application-specific dependencies or requirements. 
+
+In this tutorial, I will be showing you how to use Storybook to create a simple react application. The application will be a simple unit conversion app, and I will use Storybook to showcase the individual components and the application page itself. I will not assume any prior knowledge of React or Storybook. I will assume that you are familiar with JavaScript and Node and have an up-to-date version of the `npm` packet manager installed on your computer.
+
+
+**Prerequisites**:
+
+- [Node 14](https://nodejs.org/)
+- [Okta CLI](https://cli.okta.com)
+
+**Table of Contents**{: .hide }
+* Table of Contents
+{:toc}
+
+## Creating React components with Storybook
+
+In this section, I will show you how to create a React application and implement components displayed in Storybook. These components will serve as the basis for the unit conversion application. To start, open a terminal in a folder of your choice and run the following command to create a new React application.
+
+```bash
+npx create-react-app@5 react-storybook --use-npm
+```
+
+The `create-react-app` command creates a new folder, `react-storybook`, and initialise a basic application skeleton. Next, turn this basic React app into a Storybook application. Navigate into the newly created folder and run the following command.
+
+```bash
+npx sb@6 init
+```
+
+When prompted, answer yes to install the `sb` package. Initializing Storybook will create a new folder, `stories` inside the `src` folder, and populate it with some pre-made demo components and stories to be used by Storybook. Open the project folder in your favourite IDE. 
+
+You can test out storybook straight away. Open a terminal session in the project folder and run the following command.
+
+```bash
+npm run storybook
+```
+
+The command runs the Storybook app and opens a browser tab (http://localhost:6006). For now, you will only see the components that Storybook installs by default. You can keep Storybook running while you develop your app.
+
+Using your IDE, create a new file named `src/stories/Components.jsx`. This will be the module that will contain some basic UI components. For the sake of this tutorial, I will place all these components into a single module. In practice, you might want to spread them out over several files. Open `src/stories/Components.jsx` and paste the following code into it.
```

Comment:
```suggestion
Using your IDE, create a new file named `src/stories/Components.jsx`. This will be the module that will contain some basic UI components. For the sake of this tutorial, I will place all these components into a single module. In practice, you might want to spread them out over several files. Open `src/stories/Components.jsx` and paste in the following code.
```
avoids the dangling "it"